### PR TITLE
[DOCS-6753] Add OP quota transform doc

### PR DESCRIPTION
**Pull request by maycmlee**
### What does this PR do? What is the motivation?

Adds a OP quota transform doc.

[DOCS-6753](https://datadoghq.atlassian.net/browse/DOCS-6753)

### Merge instructions

- [x] Please merge after reviewing

### Additional notes

[DOCS-6753]: https://datadoghq.atlassian.net/browse/DOCS-6753?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+       max: 200000
+   window:
+     1m
+sinks:
+ archive_dropped:
+   type: datadog_archives
+   inputs:
+     - quota_archiving_example.dropped
+    bucket: "X"
+    service: "X"
```
Comment (maycmlee) [inline content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:197]: Need to update the `X`s

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  text: "Learn more about configuring monitors"
+---
+
+{{< callout url="https://docs.google.com/forms/d/e/1FAIpQLSfnNnV823zAgOCowCYuXJE5cDtRqIipKsYcNpaOo1LKpGfppA/viewform" btn_hidden="false" header="Request Access!">}}
+The <code>quota</code> transform is in private beta.
+{{< /callout >}}
+
+## Overview
+
+The Observability Pipelines `quota` transform allows you to set a limit on the volume of data sent to a destination within a specific timeframe. This can safeguard you from unexpected data surges that might impact your operating costs. With the transform, you can set up different ways of handling the data when the quota has been reached. For example, you can:
```
Comment (barieom) [inline content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:25]: let's do "set a limit on the volume of data or number of events"

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+- Reroute the data sent after the quota limit to another destination, such as a cold storage.
+- Sample the data sent after the quota limit so that less data is sent to the destination.
+- Drop the data sent after the quota limit.
+- Set up a monitor to alert you that the quota has been reached.
```
Comment (barieom) [inline content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:30]: let's move this to the top of the bullet list, and say "Set a 'soft' limit by setting up a monitor to alert you that the quota has been reached."

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+1. Navigate to [Observability Pipelines][1].
+1. Click on your pipeline.
+1. Click **Edit draft**.
+1. Click **+ Add Component**.
+1. Click the **Transforms** tab.
+1. Click the **Quota** tile.
+1. Enter a name for the component.
+1. Select the input(s) for the transform.
+1. In the **Limits** section:  
+    a. Select the unit type.
```
Comment (barieom) [inline content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:51]: let's do: "Select the unit type. The unit of the quota limit can be number of events or volume of data"

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+1. In the **Limits** section:  
+    a. Select the unit type.  
+    b. Enter the limit in the **Max** field.
+1. Enter the timeframe in the **Window** field.  
+    For example, if you set:
+    - **Bytes** as the unit type  
+    - `2000000000`in the **Max** field  
+    - `24h` in the **Window** field  
+    Then the transform is configured to send up to 2GB of logs per day to the destination.
+1. Click **Save**.
```
Comment (barieom) [inline content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:59]: Let's add a final step on how to handle `[quota_id].dropped`. Let's start by explaining that data coming into the transform after the threshold is met gets routed to input ID of `[quota_id].dropped`, not `[quota_id]`. could be something like:

1. Click into a downstream component to decide where to route the data from the `quota` transform before _and_ after the quota is met. The input ID for data after the limit is met is `[quota_id].dropped`

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md (hunk 1 above)
Comment (barieom) [reply content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:197]: Generalizing it with environment variable like `${DD_ARCHIVES_BUCKET}` should suffice here

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+1. In the **Notify your team** section:  
+    a. Enter a name for your monitor.  
+    b. Enter a notification message. See [Notifications][7] and [Variables][8] for more information on customizing your message.  
+    c. Select who and which services the notifications are sent to.
+1. Optionally, you can set [renotifications][9], tags, teams, and a [priority][10] for your monitor. You can also define [permissions][11] and audit notifications.
+1. Click **Create**.
+
+## Route logs sent after the limit to datadog_archives
+
+The Observability Pipelines `datadog_archives` destination formats logs into a Datadog-rehydratable format and then routes it to [Log Archives][12].
```
Comment (barieom) [inline content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:158]: Let's add:

"To set up `datadog_archives` and configure the bucket correctly, read the guide on [Routing Logs in Datadog-Rehydratable Format to Amazon S3](https://docs-staging.datadoghq.com/may/op_datadog_archives/observability_pipelines/guide/route_logs_in_datadog_rehydratable_format_to_amazon_s3/?tab=docker)

**Review by barieom — APPROVED** [review]
looks great!! Let's try to get this out EOD after addressing the comments

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md (hunk 5 above)
Comment (maycmlee) [reply content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:59]: updated

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+
+```
+
+See [Configurations][4] for more information about the sources, transforms, and sinks in the example.
+
+## Set up a monitor to alert when the quota is met
+
+### Quota metrics
+
+The following `quota` transform metrics are available:
```
Comment (kayayarai) [inline content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:115]:
```suggestion
You can use the following `quota` transform metrics to set up monitors:
```

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+
+- `quota_limit_events` (gauge)
+- `quota_limit_bytes` (gauge)
+- `component_errors_total` (counter)
+
+For the previous [example configuration](#handling-data-sent-after-the-limit), the following metric and tag combination shows all events that were sent after the limit and that were dropped.
```
Comment (kayayarai) [inline content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:121]:
```suggestion
For the previous [example configuration](#handling-data-sent-after-the-limit), the use following metric and tag combination to find all events sent after the limit and dropped.
```

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+
+- `vector.component_sent_event_bytes_total` (metric)
+    - `component_id:quota_example` (tag)
+    - `output:dropped` (tag)
```
Comment (kayayarai) [inline content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:125]:
```suggestion
- Metric: `vector.component_sent_event_bytes_total`
    - Tags: `component_id:quota_example` and `output:dropped`
```

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+
+If the `event` type is specified in the configuration, then this is the combination of metric and tags that shows all events that were sent after the limit.
```
Comment (kayayarai) [inline content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:127]:
```suggestion
If the configuration specifies the `event` type, use the following metric and tag combination to show all events that were sent after the limit.
```

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+
+- `vector.component_sent_events_total` (metric)
+    - `component_id:quota_example` (tag)
+    - `output:dropped` (tag)
```
Comment (kayayarai) [inline content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:131]:
```suggestion
- Metric: `vector.component_sent_events_total`
    - Tags: `component_id:quota_example` and `output:dropped`
```

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md [hunk 12]
```diff
@@ -0,0 +1,213 @@
...
+### Set up a metric monitor
+
+To set up a monitor to alert when the quota is reached:
+
+1. Navigate to the [New Monitor][5] page.
+1. Select **Metric**.
+1. Leave the detection method as **Threshold Alert**.
+1. In the **Define the metric** field:  
+    a. Enter `vector.component_sent_event_bytes_total` for the metric.  
+    b. In the **from** field, add `component_id:<transform_name>,output:dropped` and replace `<transform_name>` with the name of your `quota` transform.
```
Comment (kayayarai) [inline content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:142]:
```suggestion
    b. In the **from** field, add `component_id:<transform_name>,output:dropped` where `<transform_name>` is the name of your `quota` transform.  
```

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md [hunk 13]
```diff
@@ -0,0 +1,213 @@
...
+    b. Enter `1` for the **Alert threshold**. This means that if the metric query is greater than 1, then the monitor alerts.
+See [Metric Monitors][6] for more information.
+1. In the **Notify your team** section:  
+    a. Enter a name for your monitor.  
+    b. Enter a notification message. See [Notifications][7] and [Variables][8] for more information on customizing your message.  
+    c. Select who and which services the notifications are sent to.
+1. Optionally, you can set [renotifications][9], tags, teams, and a [priority][10] for your monitor. You can also define [permissions][11] and audit notifications.
+1. Click **Create**.
+
+## Route logs sent after the limit to datadog_archives
```
Comment (kayayarai) [inline content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:156]:
```suggestion
## Route logs sent after the limit to `datadog_archives`
```

File: content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md [hunk 14]
```diff
@@ -0,0 +1,213 @@
...
+
+The Observability Pipelines `datadog_archives` destination formats logs into a Datadog-rehydratable format and then routes it to [Log Archives][12]. See [Route Logs in Datadog-Rehydratable Format to Amazon S3][13] to set up `datadog_archives`.
+
+The example configuration below is similar to the previous [example configuration](#handling-data-sent-after-the-limit). The only difference is that in the example below, the destination type is `datadog_archives` so that all logs sent to Observability Pipelines, after the quota has been reached, are routed to the archives.
```
Comment (kayayarai) [inline content/en/observability_pipelines/guide/set_quotas_for_data_sent_to_a_destination.md:160]:
```suggestion
The example configuration below is similar to the previous [example configuration](#handling-data-sent-after-the-limit), except the destination type is `datadog_archives`. All logs sent to Observability Pipelines after the quota is reached routed to the archives.
```

**Review by kayayarai — CHANGES_REQUESTED** [review]

**Review by kayayarai — APPROVED** [review]

**Pull request merged**